ais/src/test/java/au/gov/amsa/ais/message/AisBStaticDataReportPartBTest.java

```diff
@@ -0,0 +1,175 @@
+		assertEquals(typeOfShip, result);
+	}
+	
+	@Test
+	public void testTypeOfShipAndCargoTypeNotAvailable() {
+		
+		final Integer typeOfShip = Integer.valueOf(0);
+		
+		AisExtractor ex = createMock(AisExtractor.class);
+		expect(ex.getValue(anyInt(), anyInt())).andReturn(typeOfShip).atLeastOnce();
+		replay(ex);
+		Integer result = AisBStaticDataReportPartB.extractShipType(ex);
+		assertNull(result);
+	}
+	
+	@Test
+	public void testCallSignNotAvailable() {
+		
+		final String callSign = "@@@@@@@";
+		
+		AisExtractor ex = createMock(AisExtractor.class);
+		expect(ex.getString(anyInt(), anyInt())).andReturn(callSign).atLeastOnce();
+		replay(ex);
+		String result = AisBStaticDataReportPartB.extractCallSign(ex);
+		assertNull(result);
+	}
+	
+	@Test
+	public void testVendorManufacturerId() {
+		final String vendorManufacturerId = "ABC";
+		
+		AisExtractor ex = createMock(AisExtractor.class);
+		expect(ex.getString(anyInt(), anyInt())).andReturn(vendorManufacturerId).atLeastOnce();
+		replay(ex);
+		String result = AisBStaticDataReportPartB.extractVendorManufacturerId(ex);
+		assertEquals(vendorManufacturerId, result);
+	}
+	
+	@Test
+	public void testVendorUnitModelCode() {
+		final Integer vendorUnitModelCode = Integer.valueOf(123);
+		
+		AisExtractor ex = createMock(AisExtractor.class);
+		expect(ex.getValue(anyInt(), anyInt())).andReturn(vendorUnitModelCode).atLeastOnce();
+		replay(ex);
+		Integer result = AisBStaticDataReportPartB.extractVendorUnitModelCode(ex);
+		assertEquals(vendorUnitModelCode, result);
+	}
+	
+	@Test
+	public void testVendorUnitSerialNumber() {
+		final Integer vendorUnitSerialNumber = Integer.valueOf(123);
+		
+		AisExtractor ex = createMock(AisExtractor.class);
+		expect(ex.getValue(anyInt(), anyInt())).andReturn(vendorUnitSerialNumber).atLeastOnce();
+		replay(ex);
+		Integer result = AisBStaticDataReportPartB.extractVendorUnitSerialNumber(ex);
+		assertEquals(vendorUnitSerialNumber, result);
+	}
+	
+	@Test
+	public void testCallSign() {
+		final String vendorCallSign = "ABC";
+		
+		AisExtractor ex = createMock(AisExtractor.class);
+		expect(ex.getString(anyInt(), anyInt())).andReturn(vendorCallSign).atLeastOnce();
+		replay(ex);
+		String result = AisBStaticDataReportPartB.extractCallSign(ex);
+		assertEquals(vendorCallSign, result);
+	}
+	
+	@Test
+	public void testDimensionA() {
+		final int dimensionA = 123;
+		
+		AisExtractor ex = createMock(AisExtractor.class);
+		expect(ex.getValue(anyInt(), anyInt())).andReturn(dimensionA).atLeastOnce();
+		replay(ex);
+		int result = AisBStaticDataReportPartB.extractDimensionA(ex);
+		assertEquals(dimensionA, result);
+	}
+	
+	@Test
+	public void testDimensionB() {
+		final int dimensionB = 123;
+		
+		AisExtractor ex = createMock(AisExtractor.class);
+		expect(ex.getValue(anyInt(), anyInt())).andReturn(dimensionB).atLeastOnce();
+		replay(ex);
+		int result = AisBStaticDataReportPartB.extractDimensionB(ex);
+		assertEquals(dimensionB, result);
```

Comment:
use verify as well in all EasyMock tests that you have created?

Reply:
Fixed